ci/Jenkinsfile-test

```diff
@@ -0,0 +1,45 @@
+pipeline {
+    agent any
+    options {
+        disableConcurrentBuilds()
+        skipDefaultCheckout()
+    }
+
+    environment {
+        PIP_INDEX_URL='https://artifacts.internal.inmanta.com/inmanta/dev'
+    }
+
+    stages {
+        stage("clear workspace and checkout source code") {
+            steps {
+                deleteDir()
+                checkout scm
+            }
+        }
+        stage("setup virtualenv") {
+            steps {
+                sh'''
+                    python3 -m venv env
+                    env/bin/pip install -U pip poetry pytest
```

Comment:
Why not use a `requirements.txt/requirements.dev.txt` file?

Reply:
I'm not sure what to do with the poetry dependency, is it a normal or a dev dependency?

Reply:
Good question. I would call it a dev dependency, because the actual application doesn't need it. But I don't have a strong opinion on this.